komodo/check_conflicts.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import pathlib
+import subprocess
+import sys
+import tempfile
+import platform
+import venv
+import contextlib
+
+import yaml
+
+from komodo.check_up_to_date_pypi import get_pypi_packages
+from komodo.package_version import strip_version
+
+
+@contextlib.contextmanager
+def temporary_environment():
+    class EnvBuilder(venv.EnvBuilder):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.context = None
+
+        def post_setup(self, context):
+            self.context = context
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        builder = EnvBuilder(with_pip=True)
+        builder.create(str(tmp_dir))
+        context = builder.context
+        yield context.env_exe
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Checks if pypi packages are compatible"
+    )
+    parser.add_argument(
+        "release_file",
+        type=lambda arg: arg
+        if pathlib.Path(arg).is_file()
+        else parser.error("{} is not a file".format(arg)),
+        help="Release file you would like to check dependencies on.",
+    )
+    parser.add_argument(
+        "repository_file",
+        type=lambda arg: arg
+        if pathlib.Path(arg).is_file()
+        else parser.error("{} is not a file".format(arg)),
+        help="Repository file where the source of the packages is found",
+    )
+    parser.add_argument(
+        "--output-file",
+        type=str,
+        default=False,
+        help="Output file where requirements are written",
+    )
+    parser.add_argument(
+        "--try-pip",
+        action="store_true",
+        help="If given, will try to install packages and check validity. Note that "
+        "creates a virtual environment and is quite slow",
+    )
```

Comment:
Do I need to pass `--try-pip` to actually have anything checked? What happens if this is not provided?

Reply:
Either way, I think invoking this program like so

    komodo-check-conflicts bleeding.yml repository.yml

should

1. if it finds conflict, exit with a non-zero exit code and output these to stdout;
2. if no conflict, exit with 0 without any output.

The parser is called "_Checks if pypi packages are compatible_" and the command is called "**komodo-check-conflicts**", so introducing "output", "requirements", and "try-pip" as default false is a bit unexpected.

Maybe remove `--try-pip`, rename `--output-file` to `--generate-requirements`. The latter should also

1. have help text explaining that the program will "Output installed packages in pip requirements format.";
2. not require a file, but generate the requirements on stdout. If the user needs something in a file, the unix philosophy warrants: 


`komodo-check-conflicts --generate-requirements > requirements.txt`

Reply:
Yes you do, will remove that option so it always checks.